### ISSUE TITLE
es6 + angu 1.5.5 + babel = exception

### ISSUE THREAD

**Issue by cybermerlin** [angular/angular.js#14676]
https://docs.angularjs.org/error/ng/areq?p0=module&p1=not%20a%20function,%20got%20Object

**View/Core.js**
```
'use strict';

/**
 * @class Core.View.Core
 * @altClassName CoreView
 * @singleton
 */
class CoreView {
	constructor() {
		console.log('Start Core.View.Core');
        }
}
CoreView.$inject = [];

export default CoreView
```

**Core.js**
```
'use strict';

import CoreView from './View/Core';
import angular from 'angular';


export default angular.module('Core', ['winjs', 'angular-winjs'])
		.controller('Core.View.Core', CoreView)
```

**package.json**
```
    "angular": "^1.5.5",
    "angular-sanitize": "^1.5.5",
    "angular-winjs": "^4.4.0",
    "babel-core": "^6.5.1",
    "babel-loader": "^6.2.2",
    "babel-preset-es2015": "^6.5.0",
    "babel-preset-stage-0": "^6.5.0",
    "babel-runtime": "^6.9.0",
```
**webpack.config.js**
```
			{
				// JS LOADER
				// Reference: https://github.com/babel/babel-loader
				// Transpile .js files using babel-loader
				// Compiles ES6 and ES7 into ES5 code
				test: /\.js$/,
				loader: 'babel',
				exclude: /(node_modules|.idea|tests|out|docs)/,
				query: {
					// https://github.com/babel/babel-loader#options
					cacheDirectory: true,
					presets: ['es2015', 'stage-0']
				}
			},
```

**Comment by gkalpak** [comment]
There is not enough info here to investigate.

What is the context ?
Where is the app run ?
How is it bundled/served ?
What version of Angular are you using ?
What dependencies do you have ?
What does the error stack-trace look like ?
What code is involved ?
How can we reproduce the problem ?

**Comment by cybermerlin** [comment]
тут же все написано вроде.
в шапке указана версия
указана какая именно ошибка, а именно: angu жалуется, что ей передали Объект вместо Функции!

предоставил оба файла с классом и подключением к модулю.

Как воспроизвести Вам эту ошибку? - да просто добавить в любой Ваш проект эти два файла... 
Вообще я уже отказался от использования этого пакета.
Так что если не хотите помощи - занимайтесь дальше отписками.

**Comment by cybermerlin** [comment]
```
"Error: [$injector:modulerr] Failed to instantiate module function CoreView() due to:
TypeError: Cannot call a class as a function
    at _classCallCheck (http://localhost:8080/dev.js:57808:100)
    at CoreView (http://localhost:8080/dev.js:57817:3)
    at Object.invoke (http://localhost:8080/dev.js:12916:20)
```

**Comment by gkalpak** [comment]
There is (still) not enough info to be sure, but it seems like we are failing to detect you are using a class and not a plain old constructor function.

Could be a browser issue, a transpiler issue or an issue with Angular.
I can't really say without more info, sorry.

**Comment by lgalfaso** [comment]
Without some extra context, it is hard to know what is going on, so I will make my best guess. I think the issue is the following, Babel here https://github.com/babel/babel/blob/master/packages/babel-helpers/src/helpers.js#L95 is trying to check that their transformed `class` was called with `new`. We fail to detect that this is a class (well... because it is not). If this is the case, then adding `CoreView.$$ngIsClass = true` should fix this.

**Comment by thorn0** [comment]
@lgalfaso This `$$ngIsClass` thing hasn't been released yet.

**Comment by lgalfaso** [comment]
@thorn0 you are right, then the alternatives would be to wait for 1.6 to be out or for Babel not to do this check (do not know if there is a flag to control this or not). One way or another, I think that Angular is doing the right thing here.

Unless somebody has an objection, I would like to close this issue.

**Comment by gkalpak** [comment]
Since we are considering removing the "pre-assign bindings on controller instance before instantiating" feature in 1.6 (or making opt-in), it would be a good idea to make it opt-out in 1.5.x (so we could get feedback from developers using it and fix any issues, before making it "mainstream").

That would also solve this issue.

**Comment by cybermerlin** [comment]
I'm now is busy with other projects, be sure to accomplish your goal in a couple of days

**Comment by gkalpak** [comment]
Although support for disabling bindings pre-assignment has landed in master (with #15095), I don't see how this is related to this issue. I don't know what I was thinking in https://github.com/angular/angular.js/issues/14676#issuecomment-222377586 :confused: 

In any case, I agree with https://github.com/angular/angular.js/issues/14676#issuecomment-222307644; there is nothing more Angular can do. Closing...
